File: redis-database.rb
#!/usr/bin/ruby

# Install Redis server and then start/stop and take backup.
#only allow root user to execute the script.

if Process::euid==0
  print "Allowed to execute.\n"
else
  print "Access denied. Execute as Sudoer.\n"
end

#Defining functions 
def get_server_install()
  print "enter the option {yes} OR {no} to install Redis server:"
  install = $stdin.gets.chomp
  case install
  when "yes"
    output1 = `sudo apt install redis`
    output2 = `redis-server --version | awk '{print $1 $2 $3}'`.strip
    print ("Redis server is installed and #{output2} \n" )
  when "no"
    print "Redis server is not installed.... \n"
  else
    print "invalid input \n"
  end 
end

def get_server_start()
  print "enter the option 'yes' to install Redis server:"
  start = $stdin.gets
  if start == "yes"
    output1 = `sudo systemctl start redis`
    output2 = `sudo systemctl status redis`
    print "Redis server is started... \n"
    print "#{output2.strip} \n"
  else
    print "invalid input \n"  
  end
end  

def get_server_stop()
   print "enter the option 'yes' to stop Redis server:"
  stop = $stdin.gets
  if stop == "yes"
    output1 = `sudo systemctl stop redis`
    output2 = `sudo systemctl status redis`
    print "Redis server is stopped... \n"
    print "#{output2.strip} \n"
  else
    print "invalid input \n"  
  end
end

def get_server_backup()
   print "enter the option 'yes' to backup Redis server:"
   backup = $stdin.gets
  if backup == "yes"
    status = `systemctl is-active redis`.strip
    print "Redis server status #{status} \n"
    if status == "active"
      output1 =`redis-cli BGSAVE`
      print "Redis server is ACTIVE and #{output1.strip} \n"
    else
      print "Redis server is NOT ACTIVE..\n"
      print "backup operation is failed.... \n"  
    end
  else
    print "invalid input \n"  
  end
end

#calling functions

case ARGV[0]
when "install"
  get_server_install()
when "start"
  get_server_start()
when "stop"
  get_server_stop()
when "backup"
  get_server_backup()
else
  print "Help: ./system-metric.rb {install|start|stop|backup}\n"
end 

File: redis-database.py
#!/usr/bin/python3

# Install Redis server and then start/stop and take backup.
#only allow root user to execute the script.

# import module.
import os
import sys
import subprocess

#only allow root user to execute the script.
if os.geteuid()==0:
    print('Allowed to execute.')
else:
    print('access denied.Execute as Sudoer.')

#Defining functions 
def get_server_install():
  print('enter the option {yes} OR {no} to install Redis server:')
  install = input()   
  if install == 'yes':
    output1 = subprocess.run(["sudo","apt","install","redis"],check=True)
    print('Redis server is installed and version is')
    output2 = subprocess.run(["redis-cli","--version"],check=True)
  else:
    print('Redis server not installed and give valid input') 

def get_server_start():
  print('enter the option {yes} to start Redis server:')
  start = input()
  if start == 'yes':
    try:
      subprocess.run(['sudo', 'systemctl', 'start', 'redis-server'], check=True)
      print("Redis server started...")
      subprocess.run(['sudo', 'systemctl', 'status', 'redis-server'], check=True)
    except subprocess.CalledProcessError as e:
      print(f"An error occurred during the installation process: {e}")
      exit(1)
  else:
    print('invalid input')

def get_server_stop():
  print('enter the option {yes} to stop Redis server:')
  stop = input()
  if stop == 'yes':
    try:   
      subprocess.run(['sudo', 'systemctl', 'stop', 'redis-server'], check=True)
      print("Redis server stopped...")
      subprocess.run(['sudo', 'systemctl', 'status', 'redis-server'], check=True)
    except subprocess.CalledProcessError as e:
      print(f"An error occurred during the installation process: {e}")
      exit(1)
  else:
    print('invalid input')

def get_server_backup():
  print('enter the option {yes} to backup Redis server:')
  backup = input()
  if backup == 'yes':
    output = subprocess.run(["sudo","systemctl","is-active","redis"], stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True,check=True)
    print(f"Redis server status is {output.stdout.strip()} ")
    value = output.stdout.strip()
    if value == 'active':
      output1 = subprocess.run(["redis-cli","BGSAVE"],check=True)
      print("Redis serevr is backed up successfully")
    else:
      print("Redis server is NOT ACTIVE and backup operation is failed")
  else:
    print('invalid input')


#calling functions
if sys.argv[1]=='install':
  get_server_install()
elif sys.argv[1]=='start':
  get_server_start()
elif sys.argv[1]=='stop':
  get_server_stop()
elif sys.argv[1]=='backup':
  get_server_backup()
else:
  print('Help: ./system-metric.rb {install|start|stop|backup}')